renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -2816,6 +2816,81 @@ void WrappedOpenGL::glCompressedMultiTexImage1DEXT(GLenum texunit, GLenum target
   }
 }
 
+void WrappedOpenGL::StoreCompressedTexData(ResourceId texId, GLenum target, GLint level,
+                                           GLint xoffset, GLint yoffset, GLint zoffset,
+                                           GLsizei width, GLsizei height, GLsizei depth,
+                                           GLenum format, GLsizei imageSize, const void *pixels)
+{
+  bool dataStored = false;
+  byte *unpackedPixels = NULL;
+  byte *srcPixels = NULL;
+  GLint unpackbuf = 0;
+
+  m_Real.glGetIntegerv(eGL_PIXEL_UNPACK_BUFFER_BINDING, &unpackbuf);
+
+  if(unpackbuf == 0 && pixels != NULL)
+  {
+    PixelUnpackState unpack;
+    unpack.Fetch(&m_Real, false);
+
+    if(unpack.FastPathCompressed(width, height, depth))
+      srcPixels = (byte *)pixels;
+    else
+      srcPixels = unpackedPixels =
+          unpack.UnpackCompressed((byte *)pixels, width, height, depth, imageSize);
+  }
+
+  if(unpackbuf != 0)
+    srcPixels = (byte *)m_Real.glMapBufferRange(eGL_PIXEL_UNPACK_BUFFER, (GLintptr)pixels,
+                                                imageSize, eGL_MAP_READ_BIT);
+
+  if(srcPixels)
+  {
+    // Only the trivial case is handled yet.
+    if(xoffset == 0 && yoffset == 0)
+    {
+      if(target == GL_TEXTURE_2D || target == GL_TEXTURE_CUBE_MAP_POSITIVE_X ||
+         target == GL_TEXTURE_CUBE_MAP_NEGATIVE_X || target == GL_TEXTURE_CUBE_MAP_POSITIVE_Y ||
+         target == GL_TEXTURE_CUBE_MAP_NEGATIVE_Y || target == GL_TEXTURE_CUBE_MAP_POSITIVE_Z ||
+         target == GL_TEXTURE_CUBE_MAP_NEGATIVE_Z || target == GL_TEXTURE_2D_ARRAY ||
+         target == GL_TEXTURE_CUBE_MAP_ARRAY)
+      {
+        if(depth <= 1)
+        {
+          size_t compressedImageSize = GetCompressedByteSize(width, height, 1, format);
+          RDCASSERT(compressedImageSize == (size_t)imageSize);
+          size_t startOffset = compressedImageSize * zoffset;
+          auto &cd = m_Textures[texId].compressedData;
+          auto &cdData = cd[target][level];
+          cdData.resize(startOffset + imageSize);
+          memcpy(cdData.data(), srcPixels, startOffset + imageSize);
+          dataStored = true;
+        }
+      }
+      else if(target == GL_TEXTURE_3D)
+      {
+        if(zoffset == 0)
+        {
+          RDCASSERT(GetCompressedByteSize(width, height, depth, format) == (size_t)imageSize);
+          auto &cd = m_Textures[texId].compressedData;
+          auto &cdData = cd[target][level];
+          cdData.resize(imageSize);
+          memcpy(cdData.data(), srcPixels, imageSize);
+          dataStored = true;
+        }
+      }
+    }
+
+    if(unpackbuf != 0)
+      m_Real.glUnmapBuffer(eGL_PIXEL_UNPACK_BUFFER);
+  }
+
+  SAFE_DELETE_ARRAY(unpackedPixels);
+
+  if(!dataStored)
+    RDCDEBUG("StoreCompressedTexData: Compressed data is not stored! (%d)", texId);
```

Comment:
btw `ResourceId` is a 64-bit value, so `%d` won't print the right figure. You can use `%llu` since I use my own printf routine so it's guaranteed to be portable (I know in default C it needs some macro like `PRIu64`).